src/DocumentDB.ChangeFeedProcessor/FeedProcessing/ChangeFeedObserver.cs

```diff
@@ -0,0 +1,58 @@
+﻿// ----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  Licensed under the MIT license.
+// ----------------------------------------------------------------
+
+namespace Microsoft.Azure.Documents.ChangeFeedProcessor.FeedProcessing
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor.Exceptions;
+
+    internal class ChangeFeedObserver : IChangeFeedObserver
```

Comment:
Name is not descriptive. Consider better name like ObserverExceptionWrappingChangeFeedObserverDecorator